src/main/java/org/jenkinsci/plugins/gitclient/JGitAPIImpl.java

```diff
@@ -47,6 +47,7 @@
 
 import javax.annotation.Nullable;
 
+import jenkins.model.Jenkins;
```

Comment:
@ndeloof 

stephenc: > I also believe given the usage of this class in [this method](https://github.com/jenkinsci/git-client-plugin/pull/212/files#diff-88af64f5dbd015aa6e8f00f355f34700R146) the intent is that the JGitAPIImpl is supposed to be instantiated on the agent and not on the master. As such I would expect to see a `writeReplace` method that replaces `this` with `Channel.export(GitAPI.class,this)` in those cases where `Channel.current() != null`... further as I suspect that the credentials conversion in this class is probably written to assume that it is being invoked on the master JVM it may be necessary to wrap the exported proxy object in a delegating `Serializable` hand rolled proxy that unencrypts the credentials before passing through to the remoting proxy so that the decryption happens on the master (which is the only place that has access to the secret key)